tests/hotplug/tolerations.go

```diff
@@ -0,0 +1,124 @@
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libkubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute][Serial]VM Tolerations", decorators.SigCompute, decorators.VMLiveUpdateFeaturesGate, Serial, func() {
+	var (
+		virtClient kubecli.KubevirtClient
+	)
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		kv := libkubevirt.GetCurrentKv(virtClient)
+		kv.Spec.Configuration.VMRolloutStrategy = pointer.P(v1.VMRolloutStrategyLiveUpdate)
+		testsuite.UpdateKubeVirtConfigValue(kv.Spec.Configuration)
+	})
+
+	Context("Updating VMs tolerations", func() {
+		It("should successfully live update tolerations", func() {
+
+			By("Creating a running VM")
+			vmi := libvmifact.NewGuestless(libvmi.WithNamespace(testsuite.GetTestNamespace(nil)))
+			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunStrategy(v1.RunStrategyAlways))
+
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
+			vmi = libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("Adding tolerations")
+			tolerations := []k8sv1.Toleration{
+				{
+					Effect:            k8sv1.TaintEffectNoExecute,
+					Key:               "testTaint",
+					Operator:          k8sv1.TolerationOpExists,
+					TolerationSeconds: pointer.P(int64(300)),
+				},
+			}
+			patchSet := patch.New(patch.WithAdd("/spec/template/spec/tolerations", tolerations))
+			vm, err = patchVM(virtClient, vm, patchSet)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring the VMI has added the tolerations")
+			validateTolerations(virtClient, vmi, tolerations[0])
+
+			By("Changing the toleration")
+			tolerations[0].TolerationSeconds = pointer.P(int64(200))
+			tolerations[0].Key = "testTaint2"
+
+			patchSet = patch.New(patch.WithReplace("/spec/template/spec/tolerations", tolerations))
+			vm, err = patchVM(virtClient, vm, patchSet)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring the VMI has changed the tolerations")
+			validateTolerations(virtClient, vmi, tolerations[0])
+
+			By("Removing tolerations")
+			patchSet = patch.New(patch.WithRemove("/spec/template/spec/tolerations"))
+			vm, err = patchVM(virtClient, vm, patchSet)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring the VMI has removed the toleration")
+			Eventually(func() bool {
+				vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, k8smetav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				for _, vmiToleration := range vmi.Spec.Tolerations {
+					if equality.Semantic.DeepEqual(vmiToleration, tolerations[0]) {
+						return false
+					}
+				}
+
+				return true
```

Comment:
This seems redundant.

Reply:
why? I have to test that the toleration gets removed, or are you referring to something else?